### ISSUE TITLE
Inconsistent loss/metrics with jax backend

### ISSUE THREAD

**Issue by dkgaraujo** [keras-team/keras#20388]
Training an LSTM-based model with `mean_squared_error` loss, I got the following training results, for which the math doesn't add up: the values of the loss (MSE) and metric (RMSE) are inconsistent.

Would anyone have an insight as to what could be happening here? Thank you in advance.

<img width="1859" alt="Screenshot 2024-10-21 at 23 33 39" src="https://github.com/user-attachments/assets/f60b95bc-5e07-45c4-8cee-5e33bbcc7e0c">

**Comment by dkgaraujo** [comment]
Apologies, the data is proprietary so I cannot post a reproducible example. However, the network is relatively simple and comprises only off-the-shelf layers.

**Comment by dkgaraujo** [comment]
Update: I noted a similar thing also on tensorflow backend. Shouldn't the metric and the loss be calculated on the same data, the same way?

**Comment by dkgaraujo** [comment]
For context: found this post [here]( https://stackoverflow.com/questions/63464944/keras-loss-and-metrics-values-do-not-match-with-same-function-in-each) suggesting that kernel regularization is also used when calculating the "official" loss metric but not the validation metric. However, I couldn't find a confirmation in the documentation.

The model I am training is indeed using a L2 regularisation.

**Comment by jeffcarp** [comment]
@dkgaraujo thanks for reporting, can you see what the results are with L2 normalization off?

**Comment by jeffcarp** [comment]
One guess: it could be because the metrics are computed at a different frequency than losses.

**Comment by fchollet** [comment]
What's your code for RMSE? The most likely explanation is non-commutativity of the sqrt operation.

**Comment by dkgaraujo** [comment]
Actually I was able to track the issue. Apparently the loss (MSE) and the metric (RMSE) diverge when there is a regularization penalty added to a layer (https://keras.io/api/layers/regularizers/), because this gets added up to the loss but is not reflected in the metric. So from my perspective as an user, not a bug, but could be better documented perhaps.

@fchollet, I did not use custom RMSE code.
